doc/nrf/gsg_guides/gsg_other.rst

```diff
@@ -1,11 +1,9 @@
 .. _gsg_other:
 
+
```

Comment:
I don't think this page serves a purpose with this content. It'd probably be better to just add a sentence or two to the Quick Start page to point out that these devices don't have support for Quick Start.

Reply:
It was added by @FrancescoSer on request from @shanthanordic for those who did not click the top page and just looked at the hierarchy: 
![image](https://github.com/user-attachments/assets/2e2ecb51-c271-4f36-bf94-acae2f8e8b45)

I'd like to remove this page once the subpages are removed from the section.

Reply:
Alright, that makes sense.